### PR TITLE
Live update chain statistics #33

### LINKED ISSUE #33: When a Contributor views the homepage, and the statistics change, they can see the new statistics update in place

### PR DESCRIPTION
**Pull request by Stamates**
Resolves #33 

## Motivation

Live update of average block time and address count on homepage (chain page).

### Enhancements
* Live update of average block time and address count
* Removed unused Statistics GenServer and moved Statistics context functions back into Chain context

### CODE REVIEW

File: apps/explorer_web/assets/js/pages/chain.js [hunk 1]
```diff
@@ -51,14 +59,19 @@ export function reducer (state = initialState, action) {
...
+    addressesChannel.on('count', msg => store.dispatch({ type: 'RECEIVED_NEW_ADDRESS_COUNT', msg: humps.camelizeKeys(msg) }))
+
+    const blocksChannel = socket.channel(`blocks:new_block`)
     store.dispatch({
       type: 'PAGE_LOAD',
       transactionCount: $('[data-selector="transaction-count"]').text()
     })
     blocksChannel.join()
     blocksChannel.on('new_block', msg => store.dispatch({ type: 'RECEIVED_NEW_BLOCK', msg: humps.camelizeKeys(msg) }))
+    blocksChannel.on('average_time', msg => store.dispatch({ type: 'RECEIVED_NEW_BLOCK', msg: humps.camelizeKeys(msg) }))
```
Comment (jimmay5469) [inline apps/explorer_web/assets/js/pages/chain.js]: I think this is a remnant.

**Comment by acravenho** [comment]
@Stamates Hey John, I'm assuming that the updated statistics are only looking for real-time addresses/balances. After refreshing I would see quite a few transactions and wallets added but this was probably due to the Genesis indexer.

**Comment by jimmay5469** [comment]
@acravenho yes, the number only gets live-updated with the realtime indexer but since the catchup indexer is running in the background the number will continue to grow between live-updates.

File: apps/explorer/lib/explorer/application.ex [hunk 2]
```diff
@@ -24,7 +24,6 @@ defmodule Explorer.Application do
 
   defp configurable_children do
     [
-      configure(Explorer.Chain.Statistics.Server),
```
Comment (KronicDeth) [inline apps/explorer/lib/explorer/application.ex]: ❤️ I hated that module

File: apps/explorer/lib/explorer/chain/statistics.ex [hunk 3]
```diff
@@ -1,123 +0,0 @@
-defmodule Explorer.Chain.Statistics do
```
Comment (KronicDeth) [inline apps/explorer/lib/explorer/chain/statistics.ex]: ❤️

File: apps/explorer/test/explorer/chain/statistics/server_test.exs [hunk 4]
```diff
@@ -1,120 +0,0 @@
-defmodule Explorer.Chain.Statistics.ServerTest do
```
Comment (KronicDeth) [inline apps/explorer/test/explorer/chain/statistics/server_test.exs]: ❤️ removing such a slow and flaky test module.

File: apps/explorer_web/lib/explorer_web/controllers/chain_controller.ex [hunk 5]
```diff
@@ -1,30 +1,48 @@
...
   use ExplorerWeb, :controller
 
-  alias Explorer.Chain.{Address, Block, Statistics, Transaction}
+  alias Explorer.{Chain, PagingOptions, Repo}
+  alias Explorer.Chain.{Address, Block, Transaction}
   alias Explorer.ExchangeRates.Token
   alias Explorer.Market
-  alias ExplorerWeb.Chain
+
+  @address_count_module Application.get_env(:explorer_web, :fake_adapter) || Chain
```
Comment (KronicDeth) [inline apps/explorer_web/lib/explorer_web/controllers/chain_controller.ex]: You shouldn't capture `Application.get_env` output in module attributes.  It is capturing runtime environment in a compile time module attribute.

File: apps/explorer_web/lib/explorer_web/notifier.ex [hunk 6]
```diff
@@ -8,7 +8,11 @@ defmodule ExplorerWeb.Notifier do
   alias Explorer.ExchangeRates.Token
   alias ExplorerWeb.Endpoint
 
+  @address_count_module Application.get_env(:explorer_web, :fake_adapter) || Chain
```
Comment (KronicDeth) [inline apps/explorer_web/lib/explorer_web/notifier.ex]: You shouldn't capture `Application.get_env` output in module attributes.  It is capturing runtime environment in a compile time module attribute.

**Pull request merged**